beacon-chain/gateway/gateway.go

```diff
@@ -0,0 +1,144 @@
+package gateway
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	gwruntime "github.com/grpc-ecosystem/grpc-gateway/runtime"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/rpc/v1_gateway"
+	"github.com/prysmaticlabs/prysm/shared"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/connectivity"
+)
+
+var _ = shared.Service(&Gateway{})
+
+// Gateway is the gRPC gateway to serve HTTP JSON traffic as a proxy and forward
+// it to the beacon-chain gRPC server.
+type Gateway struct {
+	conn        *grpc.ClientConn
+	ctx         context.Context
+	cancel      context.CancelFunc
+	gatewayAddr string
+	remoteAddr  string
+	server      *http.Server
+	mux         *http.ServeMux
+
+	startFailure error
+}
+
+// Start the gateway service. This serves the HTTP JSON traffic on the specified
+// port.
+func (g *Gateway) Start() {
+	ctx, cancel := context.WithCancel(g.ctx)
+	g.cancel = cancel
+
+	conn, err := dial(ctx, "tcp", g.remoteAddr)
+	if err != nil {
+		log.WithError(err).Error("Failed to connect to gRPC server")
+		g.startFailure = err
+		return
+	}
+
+	g.conn = conn
+
+	gwmux := gwruntime.NewServeMux()
+	for _, f := range []func(context.Context, *gwruntime.ServeMux, *grpc.ClientConn) error{
+		pb.RegisterBeaconServiceHandler,
+	} {
+		if err := f(ctx, gwmux, conn); err != nil {
+			log.WithError(err).Error("Failed to start gateway")
+			g.startFailure = err
+			return
+		}
+	}
+
+	g.mux.Handle("/", gwmux)
+
+	g.server = &http.Server{
+		Addr:    g.gatewayAddr,
+		Handler: g.mux,
+	}
+	go func() {
+		if err := g.server.ListenAndServe(); err != http.ErrServerClosed {
+			log.WithError(err).Error("Failed to listen and serve")
+			g.startFailure = err
+			return
+		}
+	}()
+
+	return
+}
+
+// Status of grpc gateway. Returns an error if this service is unhealthy.
+func (g *Gateway) Status() error {
+	log.WithField("address", g.gatewayAddr).Info("Starting gRPC gateway.")
```

Comment:
why does status request log "Starting gRPC gateway."

Reply:
Oops! It should be in the start method